lib/api/test/unit/units/validateHeaders.test.js

```diff
@@ -0,0 +1,139 @@
+const { assert } = require('chai');
+const { validateHeaders } = require('../../../units/validateHeaders');
+
+describe('validateHeaders', () => {
+  describe('given matching headers', () => {
+    const res = validateHeaders(
+      {
+        headers: {
+          'content-type': 'application/json',
+          connection: 'keep-alive'
+        }
+      },
+      {
+        headers: {
+          'content-type': 'application/json',
+          connection: 'keep-alive'
+        }
+      }
+    );
+
+    it('has proper validator', () => {
+      assert.propertyVal(res, 'validator', 'HeadersJsonExample');
+    });
+
+    it('has proper real type', () => {
+      assert.propertyVal(
+        res,
+        'realType',
+        'application/vnd.apiary.http-headers+json'
+      );
+    });
+
+    it('has proper expected type', () => {
+      assert.propertyVal(
+        res,
+        'expectedType',
+        'application/vnd.apiary.http-headers+json'
+      );
+    });
+
+    it('has no errors', () => {
+      assert.deepPropertyVal(res, 'results', []);
+    });
+  });
+
+  describe('given non-matching headers', () => {
+    const res = validateHeaders(
+      {
+        headers: {
+          connection: 'keep-alive'
+        }
+      },
+      {
+        headers: {
+          'content-type': 'application/json',
+          connection: 'keep-alive'
+        }
+      }
+    );
+
+    it('has proper validator', () => {
+      assert.propertyVal(res, 'validator', 'HeadersJsonExample');
+    });
+
+    it('has proper real type', () => {
+      assert.propertyVal(
+        res,
+        'realType',
+        'application/vnd.apiary.http-headers+json'
+      );
+    });
+
+    it('has proper expected type', () => {
+      assert.propertyVal(
+        res,
+        'expectedType',
+        'application/vnd.apiary.http-headers+json'
+      );
+    });
+
+    describe('has error', () => {
+      it('for each missing header', () => {
```

Comment:
The test says for each missing header, but it doesn't verify it. The test should have more than one missing header to verify there will be more than one results for each of them.

Reply:
Agree. Provided in 0cfab2767595f42f8caf112dc61a21b354baa33c